webapp/channels/src/components/button/index.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import type {ButtonHTMLAttributes} from 'react';
+import React from 'react';
+import type {MessageDescriptor} from 'react-intl';
+import {useIntl} from 'react-intl';
+
+type ButtonProps = {
+    label: string | MessageDescriptor;
+    emphasis: 'primary' | 'secondary' | 'tertiary' | 'quaternary' | 'link';
+    size?: 'xSmall' | 'small' | 'medium' | 'large';
+    pull?: 'left' | 'right';
+    destructive?: boolean;
+    leadingIcon?: string;
+    trailingIcon?: string;
+    testId?: string;
```

Comment:
Id is 90% of the time misused. And most of what we want to do, there is an easy way to do it with React. So for the time being, let's push back on adding the id until we know there is no easier way to do it.

I feel most of what we may need the id for, we can use a reference, which is pretty powerful, and not so "dangerous".

Reply:
The main reason I've been thinking of it is for things like using `aria-labelledby` instead of `aria-label` to have two elements share a label, but that's something new I just learned about. We can definitely re-add it if we find we need it later on